src/main/java/com/blackducksoftware/integration/hub/api/component/ComponentRestService.java

```diff
@@ -0,0 +1,48 @@
+package com.blackducksoftware.integration.hub.api.component;
+
+import static com.blackducksoftware.integration.hub.api.UrlConstants.SEGMENT_API;
+import static com.blackducksoftware.integration.hub.api.UrlConstants.SEGMENT_COMPONENTS;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.List;
+
+import org.restlet.data.Method;
+
+import com.blackducksoftware.integration.hub.api.HubItemRestService;
+import com.blackducksoftware.integration.hub.api.HubRequest;
+import com.blackducksoftware.integration.hub.exception.BDRestException;
+import com.blackducksoftware.integration.hub.rest.RestConnection;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.reflect.TypeToken;
+
+public class ComponentRestService extends HubItemRestService<ComponentItem> {
+
+	private static final List<String> CODE_LOCATION_SEGMENTS = Arrays.asList(SEGMENT_API, SEGMENT_COMPONENTS);
```

Comment:
Why is this called CODE_LOCATION_SEGMENTS? It looks like this is the component lookup api, right? something like /api/components?q=...